dex/src/state.rs

```diff
@@ -887,6 +887,27 @@ impl<'a, H: QueueHeader> Queue<'a, H> {
         Ok(value)
     }
 
+    #[inline]
+    pub fn peek_back(&self) -> Option<&H::Item> {
+        if self.empty() {
+            return None;
+        }
+        Some(&self.buf[(self.header.head() + self.header.count() - 1) as usize])
+    }
+
+    #[inline]
+    pub fn pop_back(&mut self) -> Result<H::Item, ()> {
```

Comment:
shouldn't this be an option too?

Reply:
I think it should error in the case where there is no event to pop